src/features/windows-permission-usage.js

```diff
@@ -0,0 +1,339 @@
+/* global Bluetooth, Geolocation, HID, Serial, USB */
+import { DDGProxy, DDGReflect, defineProperty } from '../utils'
+
+export function init () {
+    const featureName = 'windows-permission-usage'
+
+    const Permission = {
+        Geolocation: 'geolocation',
+        Camera: 'camera',
+        Microphone: 'microphone'
+    }
+
+    const Status = {
+        Inactive: 'inactive',
+        Accessed: 'accessed',
+        Active: 'active',
+        Paused: 'paused'
+    }
+
+    const isFrameInsideFrame = window.self !== window.top && window.parent !== window.top
```

Comment:
I think this can be swapped with: https://github.com/duckduckgo/content-scope-scripts/blob/50df895796a246a0cceb2e352b25e299d4361b19/src/utils.js#L59-L65

Reply:
If my understanding is correct `isBeingFramed` return `true` when the caller is an iframe, but in the permission script we need to determine if it's an iframe inside an iframe. Can we do that with the existing methods?

Reply:
Oh sorry you're right!
ancestorOrigins will give a better response to detect this though as it will work cross origin. I'm happy to ignore for now though :)